pkg/kubelet/pod/pod_manager_test.go

```diff
@@ -17,151 +17,246 @@ limitations under the License.
+			assert.Equal(t, test.expectShouldBeMirrorPod, actualMirrorPod, "actualMirrorPod and expectShouldBeMirrorPod differ")
+
+			if test.isMirrorPod {
+				actualPod, ok = podManager.GetPodByMirrorPod(test.wantPod)
+				assert.Equal(t, test.expectShouldBeMirrorPod != nil, ok)
+				assert.Equal(t, test.expectGetPod, actualPod, "actualPod by name and expectGetPod differ")
+			}
+		})
+	}
 }
 
 func TestRemovePods(t *testing.T) {
-	mirrorPod := &v1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			UID:       types.UID("mirror-pod-uid"),
-			Name:      "mirror-static-pod-name",
-			Namespace: metav1.NamespaceDefault,
-			Annotations: map[string]string{
-				kubetypes.ConfigSourceAnnotationKey: "api",
-				kubetypes.ConfigMirrorAnnotationKey: "mirror",
+	var (
+		mirrorPod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       types.UID("mirror-pod-uid"),
+				Name:      "mirror-static-pod-name",
+				Namespace: metav1.NamespaceDefault,
+				Annotations: map[string]string{
+					kubetypes.ConfigSourceAnnotationKey: "api",
+					kubetypes.ConfigMirrorAnnotationKey: "mirror",
+				},
 			},
-		},
-	}
-	staticPod := &v1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			UID:         types.UID("static-pod-uid"),
-			Name:        "mirror-static-pod-name",
-			Namespace:   metav1.NamespaceDefault,
-			Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "file"},
-		},
-	}
-
-	expectedPods := []*v1.Pod{
-		{
+		}
+		staticPod = &v1.Pod{
 			ObjectMeta: metav1.ObjectMeta{
-				UID:         types.UID("extra-pod-uid"),
-				Name:        "extra-pod-name",
+				UID:         types.UID("static-pod-uid"),
+				Name:        "mirror-static-pod-name",
 				Namespace:   metav1.NamespaceDefault,
-				Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "api"},
+				Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "file"},
 			},
-		},
-		staticPod,
-	}
-	updates := append(expectedPods, mirrorPod)
-	podManager, _ := newTestManager()
-	podManager.SetPods(updates)
-
-	podManager.RemovePod(staticPod)
+		}
+		normalPod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				UID:       types.UID("normal-pod-uid"),
+				Name:      "normal-pod-name",
+				Namespace: metav1.NamespaceDefault,
+			},
+		}
+	)
 
-	actualPods := podManager.GetPods()
-	if len(actualPods) == len(expectedPods) {
-		t.Fatalf("Run RemovePod() error, expected %d pods, got %d pods; ", len(expectedPods)-1, len(actualPods))
+	testCase := []struct {
+		name                             string
+		podList                          []*v1.Pod
+		needToRemovePod                  *v1.Pod
+		expectPods                       []*v1.Pod
+		expectMirrorPods                 []*v1.Pod
+		expectOrphanedMirrorPodFullnames []string
+	}{
+		{
+			name:             "Remove mirror pod",
+			podList:          []*v1.Pod{mirrorPod, staticPod, normalPod},
+			needToRemovePod:  mirrorPod,
+			expectPods:       []*v1.Pod{normalPod, staticPod},
```

Comment:
```suggestion
			expectPods:       []*v1.Pod{staticPod, normalPod},
```

Reply:
Not using sorting will cause unit tests to be unstable. 
That’s why I used cmp.diff before, because it can pass sortOpt, which keeps the code concise.

Reply:
I didn't see them being unstable. Do you? 
As for cmd.diff usage - [I was against using it to compare strings](https://github.com/kubernetes/kubernetes/pull/126895#discussion_r1730824638) as far as I remember. I was actually surprised you stopped using them and started comparing pod references.

Reply:
Sorry, I misunderstood you. I will make the changes tomorrow. Thanks for your suggestion. :)

Reply:
I've made some modifications. I'm now checking the pod object using `assert.Empty(t, cmp.Diff(actual, expect), "...")` I hope I didn't misunderstand the point. If still having any further suggestions for changes, please let me know. Thank you for your review.